fix potentially invalid id strings

arizzitano opened the pull request:
Hi! This PR introduces a small change to the way id strings are assembled in `show.html`. When ids start with a numeric character, it causes the id selector to throw in many browsers, including Firefox and phantomjs (which we use in CI). To avoid this, I've switched the order of the ids + namespace strings, so the selector will always begin with a letter. I also bumped the version, but can back that out if you'd prefer to do it in a separate PR.

Thanks for your review and please let me know if you need anything else from me to help get this merged!

andy-armstrong commented:
@pdpinch here's the a11y PR we were chatting about.

pdpinch commented:
Thank you! Please remove the version number change. We usually handle that separately.

amir-qayyum-khan commented on setup.py:
```diff
@@ -16,7 +16,7 @@ def package_data(pkg, root_list):
 setup(
     name='edx-sga',
-    version='0.6.2',
+    version='0.6.3',
```
please remove this

arizzitano commented:
Done! Thanks @pdpinch and @amir-qayyum-khan.

amir-qayyum-khan commented:
i am just running tests, locally once it done i will merge it

amir-qayyum-khan commented:
@arizzitano code looks good and all tests are passing but i was wondering can you reproduce the issue on SGA master?
because i was not able to produce it in my local Firefox.

> When ids start with a numeric character, it causes the id selector to throw in many browsers, including Firefox and phantomjs (which we use in CI).

arizzitano commented:
@amir-qayyum-khan just to be clear, the error doesn't occur on page load -- it happens whenever one of the elements with a numeric id is targeted with an id selector (e.g. `document.querySelector('#12345')`. For edX, this happens when we target `edx-sga` elements within our CI suite. We have an [open ticket](https://openedx.atlassian.net/browse/TE-1919) addressing this and several builds which have erroneously failed because of it, resulting in a significant amount of lost time for developers.

Although it isn't `edx-sga`'s js throwing these errors, many browsers still do not support numeric ids and it is encouraged to preface them with a letter character for compatibility. Per [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id):
> Using characters except ASCII letters and digits, '_', '-' and '.' may cause compatibility problems, as they weren't allowed in HTML 4. Though this restriction has been lifted in HTML 5, an ID should start with a letter for compatibility.

To reproduce the errors in Firefox, open the developer console and target a numeric id either with the `$` selector or `document.querySelector` or `document.querySelectorAll`, e.g.: `$('#12345');`. It should throw a `SyntaxError`. Here's what my output looked like (OSX, FF v53.0):

![image](https://cloud.githubusercontent.com/assets/491289/25632850/32a4afb4-2f43-11e7-826d-0151a3f21a2a.png)

amir-qayyum-khan commented:
@arizzitano thanks for explanation

pdpinch commented:
Hi @arizzitano are there some tests we should run in open edX to confirm that this is working?

arizzitano commented:
Thanks @pdpinch! We've actually been running tests against the new changes within [this ongoing PR](https://github.com/edx/edx-platform/pull/15008) addressing the issues with numeric ids across `edx-platform` and can confirm it is working as expected.

The [first a11y build](https://build.testeng.edx.org/job/edx-platform-accessibility-pr/33770/) fails using `edx-sga==0.6.2` due to several `test_problem_type_a11y` failures based on numeric elements within `edx-sga` (among other elements). However, the [latest a11y build](https://build.testeng.edx.org/job/edx-platform-accessibility-pr/34036/) passes when installing the latest from `0.6.3`, since the numeric elements no longer exist. Does this answer your question?

pdpinch commented:
Great. I've pushed [the 0.6.3 update to PyPI](https://pypi.python.org/pypi/edx-sga/0.6.3). It looks like the version bump in edx will be part of your PR, so I won't open a PR on edx-platform for that.
